bokehjs/src/lib/models/widgets/time_picker.ts

```diff
@@ -0,0 +1,145 @@
+import flatpickr from "flatpickr"
+
+import {PickerBase, PickerBaseView} from "./picker_base"
+import {String, Number, Or} from "core/kinds"
+import * as p from "core/properties"
+import {assert} from "core/util/assert"
+
+export type TimeLike = typeof TimeLike["__type__"]
+export const TimeLike = Or(String, Number)
+
+export class TimePickerView extends PickerBaseView {
+  declare model: TimePicker
+
+  protected _format_time(date: Date): string {
+    const {picker} = this
+    return picker.formatDate(date, picker.config.dateFormat)
+  }
+
+  override connect_signals(): void {
+    super.connect_signals()
+
+    const {
+      value, min_time, max_time, time_format, hour_increment,
+      minute_increment, second_increment, seconds, am_pm,
+    } = this.model.properties
+
+    this.connect(value.change, () => {
+      const {value} = this.model
+      if (value != null) {
+        this.picker.setDate(value)
+      } else {
+        this.picker.clear()
+      }
+    })
+    this.connect(min_time.change, () => this.picker.set("minTime", this.model.min_time))
+    this.connect(max_time.change, () => this.picker.set("maxTime", this.model.max_time))
+    this.connect(time_format.change, () => this.picker.set("altFormat", this.model.time_format))
+    this.connect(hour_increment.change, () => this.picker.set("hourIncrement", this.model.hour_increment))
+    this.connect(minute_increment.change, () => this.picker.set("minuteIncrement", this.model.minute_increment))
+    this.connect(second_increment.change, () => this._update_second_increment())
+    this.connect(seconds.change, () => this.picker.set("enableSeconds", this.model.seconds))
+    this.connect(am_pm.change, () => this.picker.set("time_24hr", this.model.am_pm))
```

Comment:
We have called this `military_time` in Panel.

Reply:
To me "military" time is an Americanism. I'm not sure what's the best name, but I think that boolean `am_pm` is the most universal name I considered so far.

Reply:
I'm also 👎 on military time. Why not just use the same `time_24hr`?

Reply:
`time_24hr` is an option. I'm also considering an enum like this: `clock = Enum("12h", "24h", default="24h")`.

Reply:
👍 for enum

Reply:
Also a fan of the enum :slightly_smiling_face: